src/baldr/graphreader.cc

```diff
@@ -732,6 +733,28 @@ std::pair<GraphId, GraphId> GraphReader::GetDirectedEdgeNodes(const GraphTile* t
   return std::make_pair(start_node, end_node);
 }
 
+std::string GraphReader::edge_shape(const valhalla::baldr::GraphId& edgeid) {
+  if (!edgeid.Is_Valid()) {
+    return {};
+  }
+
+  const baldr::GraphTile* t_debug = GetGraphTile(edgeid);
+  if (t_debug == nullptr) {
+    return {};
+  }
+
+  const baldr::DirectedEdge* directedEdge = t_debug->directededge(edgeid);
+  if (directedEdge == nullptr) {
```

Comment:
this check is not needed because the method you call to get the edge will throw if the edge is not in bounds

Reply:
https://github.com/valhalla/valhalla/blob/master/valhalla/baldr/graphtile.h#L177

Reply:
haha, get dizzy of what throws and what not.